kyuubi-server/src/main/scala/org/apache/kyuubi/server/BackendServiceFetchRowsRate.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server
+
+import org.apache.hive.service.rpc.thrift.{TBinaryColumn, TBoolColumn, TByteColumn, TDoubleColumn, TI16Column, TI32Column, TI64Column, TRowSet, TStringColumn}
+
+import org.apache.kyuubi.metrics.{MetricsConstants, MetricsSystem}
+import org.apache.kyuubi.operation.FetchOrientation.FetchOrientation
+import org.apache.kyuubi.operation.OperationHandle
+import org.apache.kyuubi.service.BackendService
+
+trait BackendServiceFetchRowsRate extends BackendService {
+  abstract override def fetchResults(
+      operationHandle: OperationHandle,
+      orientation: FetchOrientation,
+      maxRows: Int,
+      fetchLog: Boolean): TRowSet = {
+    val rowSet = super.fetchResults(operationHandle, orientation, maxRows, fetchLog)
+    if (rowSet == null) return rowSet
+    val rowsSize =
+      if (rowSet.getColumnsSize > 0) {
+        rowSet.getColumns.get(0).getFieldValue match {
+          case t: TStringColumn => t.getValues.size()
+          case t: TDoubleColumn => t.getValues.size()
+          case t: TI64Column => t.getValues.size()
+          case t: TI32Column => t.getValues.size()
+          case t: TI16Column => t.getValues.size()
+          case t: TBoolColumn => t.getValues.size()
+          case t: TByteColumn => t.getValues.size()
+          case t: TBinaryColumn => t.getValues.size()
+          case _ => 0
+        }
+      } else {
+        rowSet.getRowsSize
+      }
```

Comment:
simply getRowsSize?

Reply:
we don't have RowsSize for column based set?

Reply:
```scala
  def toColumnBasedSet(rows: Seq[Row], schema: StructType, timeZone: ZoneId): TRowSet = {
    val size = rows.length
    val tRowSet = new TRowSet(0, new java.util.ArrayList[TRow](size))
    schema.zipWithIndex.foreach { case (filed, i) =>
      val tColumn = toTColumn(rows, i, filed.dataType, timeZone)
      tRowSet.addToColumns(tColumn)
    }
    tRowSet
  }
```

Reply:
Yes, we have `RowsSize` for column based set. And RowSet may be a row base set. So I first `if` try to get column base set rows and `else` to get row base set rows.

Reply:
I see
Is it efficient?

Or we can use the used `columnCount` for deliver this value from engine side

Reply:
If we use `columnCount` to deliver `rowSize`, it may cause some confuse, I don't think it's a good idea.
I think it's efficient, because it just a `match case`. And I do a little test, it shows that my guess is right:
```
             count = 8508349
         mean rate = 12083.92 calls/second
     1-minute rate = 16944.96 calls/second
     5-minute rate = 14413.72 calls/second
    15-minute rate = 7229.02 calls/second
               min = 0.00 milliseconds
               max = 0.00 milliseconds
              mean = 0.00 milliseconds
            stddev = 0.00 milliseconds
            median = 0.00 milliseconds
              75% <= 0.00 milliseconds
              95% <= 0.00 milliseconds
              98% <= 0.00 milliseconds
              99% <= 0.00 milliseconds
            99.9% <= 0.00 milliseconds
```